### PR TITLE
[NodeFW] Support graceful shutdown via killswitch

### PR DESCRIPTION
**Pull request by williampsmith**
## Description 

As in title.

## Test plan 

Will run locally once I figure out how

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: nre/bpf/nodefw/nodefw/src/main.rs [hunk 1]
```diff
@@ -15,12 +15,16 @@ use std::sync::{Arc, RwLock};
...
+pub const KILLSWITCH_FILENAME: &str = "__FW_KILLSWITCH__";
+
 #[derive(Debug, Parser)]
 struct Opt {
     #[clap(short, long, default_value = "lo")]
     iface: String,
     #[clap(short, long, value_enum, default_value_t=Mode::Default)]
     mode: Mode,
+    #[clap(short, long)]
+    killswitch_path: String,
```
Comment (suiwombat) [inline nre/bpf/nodefw/nodefw/src/main.rs:27]: can we rename this to drain_file.  killswitch implies a non-graceful shutdown when i read it.

File: nre/bpf/nodefw/nodefw/src/main.rs [hunk 2]
```diff
@@ -103,7 +107,36 @@ async fn main() -> Result<(), anyhow::Error> {
...
+            }
+            Err(err) => {
+                if path.exists() {
+                    panic!("Failed to read firewall killswitch file: {}", err);
+                }
+            }
+        }
+        tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+    }
+}
```
Comment (suiwombat) [inline nre/bpf/nodefw/nodefw/src/main.rs:142]: can we refactor this a bit to be more like what is in https://github.com/MystenLabs/sui/pull/17195?

it's fine if you prefer to write text to the file, but the existence of the file is more than sufficient.  Please also utilize the cancellationtoken.

**Comment by suiwombat** [comment]
forgot to mention, https://github.com/MystenLabs/sui/pull/17195 has details of how to actually test the firewall.  you'll need a linux host to build it.

**Comment by williampsmith** [comment]
Closing in lieu of https://github.com/MystenLabs/sui/pull/17195

**Pull request closed without merging**